contracts/backend.go

```diff
@@ -0,0 +1,60 @@
+package contracts
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/core/vm"
+	"github.com/celo-org/celo-blockchain/params"
+)
+
+type evmBackend struct {
+	vmConfig    vm.Config
+	vmContext   vm.Context
+	chainConfig *params.ChainConfig
+	state       vm.StateDB
+
+	dontMeterGas bool
```

Comment:
Having played with this before, I think that an `evm builder` is the right level of abstraction for making these calls. I don't have a solution for this, but I find the `dontMeterGas` that's only used for `GetRegisteredAddress` that also calls functions of the same name (except on a `evm` instead of `ev`) a little confusing. It's also dragging down a boolean flag through several layers of code.